### DISCUSSION TITLE
How to install E3SM on Perlmutter (NERSC server)?

### DISCUSSION THREAD

**Discussion by xiangli1025** [E3SM-Project/E3SM#6012] [E3SM model help]
Hi all,

I just got access to the NERSC server Perlmutter. Do I need to install E3SM on it or I could just load it from somewhere?

Thanks!
Xiang

**Comment by rljacob** [comment]
Our usage model is that you "git clone" the code and build your own executable.   It should build with no problems on perlmutter.

**Reply by xiangli1025** [reply]
Thanks Jacob! Could you please add me to the E3SM private organization?

**Reply by rljacob** [reply]
The code is public.  You don't need to be added to it.

**Reply by xiangli1025** [reply]
Hi Jacob, I'm trying to clone the repository to my local account. It looks like this. Any suggestions will be appreciated!

@perlmutter:login04:~/test231031> git clone  --recursive git@github.com:E3SM-Project/E3SM.git
Cloning into 'E3SM'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

**Reply by xiangli1025** [reply]
I also do not have access to this page (/wiki/spaces/ED/pages/1867951) to check the list of necessary software to build/run ACME.

**Reply by rljacob** [reply]
You need to add ssh keys to github to clone.  All necessary software is already on perlmutter and the build system knows how to find it.

**Reply by xiangli1025** [reply]
Thanks, Jacob!

I have generated a new SSH key and am trying to add it to the ssh-agent.

I am now doing this step "ssh-add --apple-use-keychain ~/.ssh/id_ed25519", but it looks like this:

@perlmutter:login04:~> ssh-add --apple-use-keychain ~/.ssh/id_ed25519
unknown option -- -
usage: ssh-add [-cDdKkLlqvXx] [-E fingerprint_hash] [-S provider] [-t life]
               [file ...]
       ssh-add -s pkcs11
       ssh-add -e pkcs11
       ssh-add -T pubkey ...

Otherwise, shall I do like this way? But what PIN should I enter?

@perlmutter:login04:~> ssh-add -K ~/.ssh/id_ed25519
Enter PIN for authenticator: 
Provider "internal" returned failure -1
Unable to load resident keys: invalid format

**Reply by xiangli1025** [reply]
By the way, I did not see the .bashrc file in my home directory. Is there something like this on perlmutter?

**Reply by sugerWu-2822** [reply]
> By the way, I did not see the .bashrc file in my home directory. Is there something like this on perlmutter?

Hello Xiang,

I'm planning to use the E3SM model at NERSC HPC Perlmutter recently, and I'm not sure how it works, could you give me some advice or guidance. Also, I found that Perlmutter does not have .bashrc file, did you solve this problem?

Looking forward to your reply.

Sincerely,
Xiuyi

**Reply by xiangli1025** [reply]
Hi Xiuyi,

You can use E3SM on Perlmutter by following the steps here (https://acme-climate.atlassian.net/wiki/spaces/ET/pages/4285759528/Pre-tutorial+materials).

Creating a .bashrc would work.

Best,
Xiang

**Comment by sugerWu-2822** [comment]
Many thanks!


Xiuyi











在 2024-05-29 21:40:10，"xiangli1025" ***@***.***> 写道：

Hi Xiuyi,

You can use E3SM on Perlmutter by following the steps here (https://acme-climate.atlassian.net/wiki/spaces/ET/pages/4285759528/Pre-tutorial+materials).

Creating a .bashrc would work.

Best,
Xiang
